src/renderer/hooks/usePoolCycle.ts

```diff
@@ -12,68 +12,68 @@ import { useMidgardContext } from '../contexts/MidgardContext'
+} => {
   const { mimir$, reloadMimir } = useThorchainContext()
   const {
     service: { thorchainConstantsState$, reloadThorchainConstants }
   } = useMidgardContext()
 
-  const poolCycleFromMidgardConstants$: LiveData<Error, number> = useMemo(
+  const midgardConstantsPoolCycle$: PoolCycleLD = useMemo(
     () =>
       FP.pipe(
         thorchainConstantsState$,
-        liveData.map(({ int_64_values }) => int_64_values.PoolCycle)
+        liveData.map(({ int_64_values }) => int_64_values.PoolCycle),
+        liveData.mapLeft(() => ({ errorId: ErrorId.GET_POOL_CYCLE, msg: 'Unable to load pool cycle from Midgard' }))
       ),
     [thorchainConstantsState$]
   )
 
-  const getPoolCycleFromMidgard = useCallback(() => {
-    reloadThorchainConstants()
-    return poolCycleFromMidgardConstants$
-  }, [reloadThorchainConstants, poolCycleFromMidgardConstants$])
+  const mimirPoolCycle$: PoolCycleLD = useMemo(
+    () =>
+      FP.pipe(
+        mimir$,
+        liveData.map(({ 'mimir//POOLCYCLE': poolCycle }) => O.fromNullable(poolCycle)),
+        liveData.chain(liveData.fromOption(() => Error('Unable to load pool cycle from Mimir'))),
+        liveData.mapLeft(({ message }) => ({ errorId: ErrorId.GET_POOL_CYCLE, msg: message }))
+      ),
+    [mimir$]
+  )
 
-  const getPoolCycleFromMimir = useCallback(() => {
+  const getData = useCallback(() => {
     reloadMimir()
-    return mimir$
-  }, [reloadMimir, mimir$])
+    reloadThorchainConstants()
+  }, [reloadThorchainConstants, reloadMimir])
 
-  const getData = useCallback(() => {
-    FP.pipe(
-      getPoolCycleFromMimir(),
-      liveData.chain(({ 'mimir//POOLCYCLE': newPoolCycle }) =>
-        FP.pipe(
-          O.fromNullable(newPoolCycle),
-          O.fold(
-            (): LiveData<Error, number> => Rx.of(RD.failure(Error(`No POOLCYCLE at the mimir's response`))),
-            (poolCycle) => liveData.of(poolCycle)
+  const [data] = useObservableState(
+    () =>
+      FP.pipe(
+        Rx.combineLatest([midgardConstantsPoolCycle$, mimirPoolCycle$]),
+        RxOp.map(([mimirPoolCycleRD, midgardPoolCycleRD]) =>
+          FP.pipe(
+            mimirPoolCycleRD,
+            RD.alt(() => midgardPoolCycleRD)
           )
-        )
+        ),
+        RxOp.distinctUntilChanged(eqPoolCycle.equals)
       ),
-      liveData.chainOnError(getPoolCycleFromMidgard),
-      liveData.mapLeft(() => ({
-        errorId: ErrorId.GET_MIMIR,
-        msg: 'Unable to load pool cycle'
-      }))
-    ).subscribe(setPoolCycleState)
-  }, [getPoolCycleFromMimir, setPoolCycleState, getPoolCycleFromMidgard])
-
-  const [data] = useObservableState(
-    () => FP.pipe(poolCycleState$, RxOp.shareReplay(1), RxOp.distinctUntilChanged(eqPoolCycle.equals)),
     RD.initial
   )
 
+  // Reload data on mount
   useEffect(() => {
     getData()
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  return [data, getData]
+  return useMemo(() => ({ data, getData }), [data, getData])
```

Comment:
Why `useMemo`?

Reply:
just to avoid possible re-creating of object